### ISSUE TITLE
Multiple games crash when DXVK is build with avx

### ISSUE THREAD

**Issue by random2324** [Tk-Glitch/PKGBUILDS#515]
I've spend some hours figuring out why some games do crash on start for some months while they ran before fine. Games affected are Rome: Total War, Titan Quest, Deus EX: HR, ARMA Tactics probably more. Some games are unaffected and run like always.

Usually there is something about directx9 in crash and EntryPoint (see log). Now, Rome is actually dx8 and Deus EX is dx11 so they shouldn't even have something to do with dx9.  But are still crashing.
[steam-28050.log](https://github.com/Tk-Glitch/PKGBUILDS/files/4313294/steam-28050.log)
If I use official proton 5.0 or precompiled dxvk 1.5.5 there is actually no issue. So it has something to do with how I build DXVK.

I used mingw 64 from the gcc portable stuff and edited updxvk.cfg so that it picks it up. Still no go.

After endless recompiles (first tries produced incorrect results because of CCACHE_DISABLE=1 not set) I figured out, that it has something to do with avx. The idea more or less came from: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=dxvk-winelib

So maybe it just has something to do with my system. Im on Haswell so maybe more people could test? Here is the modified patch I used:
[nativeopt.dxvkpatch.txt](https://github.com/Tk-Glitch/PKGBUILDS/files/4314542/nativeopt.dxvkpatch.txt)

**Comment by Tk-Glitch** [comment]
Building DXVK with -march=native is known to create problems (and even beyond d3d9, such as making Dirt Rally 2 crash when msaa is enabled, which would indicate alignment issues). Simply disabling AVX might not cover all cases broken by this - and we're likely unaware of many other cases.

It looks like I have never merged the warning notes about this in the .cfg, so I'll do that now.

**Comment by Tk-Glitch** [comment]
I guess adding the no-avx flag to the native opt patch sounds good if it fixes at least *some* cases. Thank you.

**Comment by random2324** [comment]
Yes if the option is really known to cause such problems it shouldnt be used. I never heard of such problems before so Im kinda surprised this is already known. Even though I had a similar problem with amdvlk driver: https://github.com/GPUOpen-Drivers/AMDVLK/issues/50

There are also two wine bugs:
https://bugs.winehq.org/show_bug.cgi?id=45289
https://bugs.winehq.org/show_bug.cgi?id=43516

So I will exclude avx from my proton build too.

Maybe this is a general bug? And also long standing giving the gcc bug is unresolved since 2011.

**Comment by Tk-Glitch** [comment]
If it is, it's sneaky enough to not create huge issues everywhere 🐸 Corner cases can be a pain.

**Comment by loathingKernel** [comment]
For me it was StarCraft II. I don't quite remember how I came across the idea to test disabling AXV for `-march=native`. Disabling it also seems to affect performance for the better regardless of the d3d version and title in the limited title I have tested this with.
